### PR TITLE
Subproject support for cpplint.

### PR DESCRIPTION
**Pull request by ptoharia**

### CODE REVIEW

File: CpplintTargets.cmake [hunk 1]
```diff
@@ -24,12 +24,12 @@ if(NOT CPPLINT_FOUND)
 endif(NOT CPPLINT_FOUND)
 
 if(NOT CPPLINT_FOUND)
-  add_custom_target(cpplint COMMENT "${CPPLINT_NOT_FOUND_MSG}")
-  set_target_properties(cpplint PROPERTIES EXCLUDE_FROM_ALL TRUE)
+  add_custom_target(cpplint_${PROJECT_NAME} COMMENT "${CPPLINT_NOT_FOUND_MSG}")
```
Comment (rdumusc) [inline CpplintTargets.cmake]: The convention is to use: ${PROJECT_NAME}_cpplint instead

Comment (rdumusc) [inline CpplintTargets.cmake]: Hmm, it looks more complex actually, I'm trying to understand why there are so many targets...

Comment (ptoharia) [inline CpplintTargets.cmake]: @rdumusc We tried to mimic what was already written for cppcheck

**Comment by eile** [comment]
Superseeded by #303

**Pull request closed without merging**